### PR TITLE
#929 Prepending player prefs keys with a hashed app secret to differentiate between different patchers on the same system.

### PR DESCRIPTION
**Pull request by Dzejkop**

### CODE REVIEW

File: Assets/PatchKit Patcher/Scripts/AppData/HashCalculator.cs [hunk 1]
```diff
@@ -27,7 +27,7 @@ public static string ComputeHashString(byte[] buffer, int offset, int length)
 
         public static string ComputeStringHash(string str)
         {
-            return string.Concat(new xxHash(Seed).ComputeHash(Encoding.UTF8.GetBytes(str)).Select(b => b.ToString("X2")));
+            return string.Concat(new xxHash(Seed).ComputeHash(Encoding.UTF8.GetBytes(str)).Select(b => b.ToString("X2")).ToArray());
```
Comment (genail) [inline Assets/PatchKit Patcher/Scripts/AppData/HashCalculator.cs]: Why this change? Wouldn't it break existing usages of ComputeStringHash()?

File: Assets/PatchKit Patcher/Scripts/AppData/Local/UnityCache.cs [hunk 2]
```diff
@@ -1,19 +1,38 @@
...
+        private readonly string _hashedSecret;
+
+        public UnityCache(string appSecret)
+        {
+            _hashedSecret = HashSecret(appSecret);
+        }
+
+        private string HashSecret(string secret)
+        {
+            return HashCalculator.ComputeStringHash(secret);
```
Comment (genail) [inline Assets/PatchKit Patcher/Scripts/AppData/Local/UnityCache.cs]: Please don't use xxHash for these kind of hashing, instead check what hashing algorithms are build-in in standard library and use it. I would recommend using MD5.

Reason: The main purpose of using xxHash to hash application's content is the hashing speed. If for some reason we will be forced to change this algorithm, it would be a problem to completely remove it from the patcher if there's an usage of this algorithm besides its main purpose.

File: Assets/PatchKit Patcher/Scripts/AppData/HashCalculator.cs (hunk 1 above)
Comment (Dzejkop) [reply Assets/PatchKit Patcher/Scripts/AppData/HashCalculator.cs]: It's something to do with the older C# version.
When I tried outputting the result of this hash it logged something along the lines of 
`System.Linq.Enumerable+<CreateSelectIterator>c__Iterator102[System.Byte,System.String]`, until I added the `ToArray()` call.
